### ISSUE TITLE
Controlled select menus not changing value in Firefox in RMWC 5.1.3

### ISSUE THREAD

**Issue by MiK546** [jamesmfriedman/rmwc#412]
I'm using RMWC 5.1.3 with React 16.8.2. I use create-react-app defaults. Firefox version is 65.0.

After updating to the latest version of RMWC from 4.0.6, controlled select menus stopped changing their value on Firefox. This seems to only happen when selecting the options using mouse. Keyboard selection is still ok. The select menus also seem to work fine in Chrome/ium. `evt.target.value` in the `onChange` callback returns the previously selected value no matter which optiuon is actually selected. When I call `setState()` on the state variable set in the select menus `value` attribute, the previous value gets set and nothing changes.

#### Example non-working code

```js
class Example extends Component{
    constructor(props){
        super(props);
        this.state = {
            stateValue: "value1"
        };
    }

    render(){
        return (
            <Select
                label="Select a value"
                value={this.props.stateValue}
                options={[
                    {
                        label: "Value 1 label",
                        value: "value1"
                    },{
                        label: "Value 2 label",
                        value: "value2"
                    }
                ]}
                onChange={(evt) => {
                    this.setState({
                        stateValue: evt.target.value
                    });
                }}
            />
        );
    }
}
```

In this example, no matter what you select on the select menu using the mouse, the `evt.target.value` will always be `value1`. This leads to the select menu not functioning.

**Comment by jamesmfriedman** [comment]
Thanks for reporting. Interesting that its only a regression in Firefox, I'll take a look

**Comment by jamesmfriedman** [comment]
Interestingly enough, its only a bug on Firefox when using the formatted options, and works correctly with a flat array or an object.

**Comment by jamesmfriedman** [comment]
Fixed in 5.1.4. Not sure exactly how deep the issue was, but it appears to be an issue with either a change in Firefox, or a Change in MDC. Either way, its fixed.
